packages/api-server/api_server/repositories/tasks.py

```diff
@@ -27,6 +28,16 @@ def __init__(self, user: User):
         self.user = user
 
     async def save_task_state(self, task_state: TaskState) -> None:
+        datetime_request_time = (
+            datetime.fromtimestamp(task_state.unix_millis_request_time / 1000)
+            if task_state.unix_millis_request_time
+            else datetime.now()
+        )
```

Comment:
The task state is published by `rmf_fleet_adapter::TaskManager`, which at the moment, does not populate the `unix_millis_request_time` field, https://github.com/open-rmf/rmf_ros2/blob/main/rmf_fleet_adapter/src/rmf_fleet_adapter/TaskManager.cpp#L459, which means we will always be setting `task_state.unix_millis_request_time` as the current time. We will need to query the previous task state of this task ID, and obtain the `previous_task_state.unix_millis_request_time` to be used here instead